bottle.py

```diff
@@ -950,18 +951,24 @@ def _cast(self, out, peek=None):
 
     def wsgi(self, environ, start_response):
         """ The bottle WSGI-interface. """
+        provided_exc_info = False
         try:
             out = self._cast(self._handle(environ))
             # rfc2616 section 4.3
             if response._status_code in (100, 101, 204, 304)\
             or environ['REQUEST_METHOD'] == 'HEAD':
                 if hasattr(out, 'close'): out.close()
                 out = []
-            start_response(response._status_line, response.headerlist)
+            exc_info = environ.get('bottle.exc_info')
+            if exc_info is not None:
+                del environ['bottle.exc_info']
+                provided_exc_info = True
+            start_response(response._status_line, response.headerlist, exc_info)
             return out
         except (KeyboardInterrupt, SystemExit, MemoryError):
             raise
         except Exception:
+            if provided_exc_info: raise
```

Comment:
When reviewing this again I noticed that the above `start_response` is never called multiple times (though the below one might be). 

I included this change because of the following line in the [error handling section](https://www.python.org/dev/peps/pep-3333/#error-handling) of pep-3333:

> Never trap errors raised by start_response when exc_info is being provided

However, if I'm reading the intent behind it correctly, it's not actually needed here. Even though we're adding `exc_info` to the above `start_response()`, the server can't have sent any data yet (it's our first call), so `start_response()` will never need to raise the type of exception we're not allowed to trap.

Removing this line (and others related to `provided_exc_info`) would avoid the possibility of edge cases where we passed `exc_info` to `start_response()` and then it raised an unrelated exception, which we then didn't catch.